jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/migration/MigrationManager.java

```diff
@@ -579,10 +584,19 @@ public Void execute(Context context) {
                         org.jbpm.workflow.instance.NodeInstance active = processInstance.getNodeInstance(entry.getKey(), false);
                         if (active instanceof TimerNodeInstance) {
                             TimerInstance timerInstance = entry.getValue().get(0);
-
+                            
                             long delay = timerInstance.getDelay() - (System.currentTimeMillis() - timerInstance.getActivated().getTime());
                             timerInstance.setDelay(delay);
-
+                            
+                            Trigger trigger = ((DefaultJobHandle)timerInstance.getJobHandle()).getTimerJobInstance().getTrigger();
+                            if (trigger instanceof IntervalTrigger && timerInstance.getPeriod() > 0) {
+                                if (timerInstance.getRepeatLimit() > 0) {
+                                    timerInstance.setRepeatLimit(((IntervalTrigger) trigger).getRepeatLimit() - ((IntervalTrigger) trigger).getRepeatCount());
+                                }
+                                delay = timerInstance.getPeriod() - (System.currentTimeMillis() - ((IntervalTrigger) trigger).getLastFireTime().getTime());
+                                timerInstance.setDelay(delay);
```

Comment:
Just to be sure, the delay will be reset to period value after the current trigger is fired, right?

Reply:
as soon as the timer is cycle based it should rely on period

Reply:
So this will create an ordinary timer and only after that a cycle based one?

Reply:
correct